### PR TITLE
lib: nrf_modem: Allow to create native sockets in parellel to offloaded

### PR DESCRIPTION
**Pull request by rlubos**

### CODE REVIEW

**Comment by rlubos** [comment]
@jhirsi SOCK_NATIVE could (or even should) also be used with packet socket in MoSH to create a native sock.

**Comment by jhirsi** [comment]
> @jhirsi SOCK_NATIVE could (or even should) also be used with packet socket in MoSH to create a native sock.

OK. Thanks for the headsup! That change could be included already in this PR by changing:
https://github.com/nrfconnect/sdk-nrf/blob/master/samples/nrf9160/modem_shell/src/ppp/ppp_ctrl.c#L224
->
`ppp_data_socket_fd = socket(AF_PACKET, (SOCK_RAW | SOCK_NATIVE), IPPROTO_RAW);`

**Comment by rlubos** [comment]
@jhirsi I have added an extra commit for `modem_shell`, I didn't test the PPP functionality though, would be nice if you could give it a shot.

**Comment by jhirsi** [comment]
> @jhirsi I have added an extra commit for `modem_shell`, I didn't test the PPP functionality though, would be nice if you could give it a shot.

Tested and socket creation works ok and ppp is working end-to-end when including thatg into your branch (not yet in ncs): https://github.com/zephyrproject-rtos/zephyr/commit/9de47d2b0b4f1c927997fb4e577a6d1dcbe95eba

File: doc/nrf/releases/release-notes-changelog.rst [hunk 1]
```diff
@@ -88,6 +88,10 @@ nRF9160
 
     * Added support for Firmware Update object to use :ref:`lib_fota_download` library for downloading firmware images.
 
+  * :ref:`_nrf_modem_lib_readme` library:
```
[the remark this answers is not in the document]
Comment (rlubos) [reply doc/nrf/releases/release-notes-changelog.rst]: Fixed, thanks.

**Review by jhirsi — APPROVED** [review]

**Pull request merged**